bench/workgen/workgen.cxx

```diff
@@ -1561,20 +1621,28 @@ Operation::is_table_op() const
 }
 
 void
-Operation::kv_compute_max(bool iskey, bool has_random)
+Operation::kv_compute_max(ThreadRunner *runner, bool iskey, bool has_random)
 {
     ASSERT(is_table_op());
     TableOperationInternal *internal = static_cast<TableOperationInternal *>(_internal);
 
-    int size = iskey ? _key._size : _value._size;
-    if (size == 0)
-        size = iskey ? _table.options.key_size : _table.options.value_size;
+    int min_size_key = 2;
+    int min_size_value = 1;
+    int size;
+    if (_random_kv) {
+        // TODO - Need a range for the key/value.
```

Comment:
@ddanderson, what about the size here? I don't have information on the key/value size for the randomly selected table, how can I know what size I should go for?

Reply:
@etienneptl There are two reasons to compute the size before the run.  The main reason is that each thread has a key buffer and a value buffer that holds the data for each operation.  Since (in the past) we knew every operation for a thread, we avoid reallocing ever while running the workload.  We just get the max size, and allocate that.  The second reason is more just convenience, since we can store the key and data size in the operation, and it's ready to go.  I think for these changes, if the size is random, you could set it to a specific sentinal, maybe -1?  (I'd rather not use 0 - perhaps someday we'll want a way to run with zero size values, at least).  For the second convenience reason if we see -1, you'll compute it at runtime, you'll need to put a table anyway (I'm curious if the random size range attached to the table or the operation's value?).  For the first reason, we'll have to bite the bullet and realloc as we go.

Just to speculate, if you had a max size for the various random tables, you could think about allocating that.  BUT if you had a pareto distribution where almost all values are 1k and one in a 10 million are 10M, we probably don't want to allocate 10M per thread.  So you may need to handle this.  If huge values are a possibility, you probably want to say something like, if I'm asked to operate with a value >100K in size, I malloc the buffer for the operation and immediately free it.  It all depends on the maximum sizing you're looking at now or in the future, and the max number of threads, and the amount of memory you're willing to give up just for data buffers.